webapp/store/models.py

```diff
@@ -187,14 +188,14 @@ def _parse_charm_data(charm_data, include_files=False):
         'options': meta.get('charm-config', {}).get('Options'),
         'owner': meta.get('owner', {}).get('User'),
         'promulgated': meta.get('promulgated', {}).get('Promulgated'),
-        'provides': meta['charm-metadata'].get('Provides'),
-        'requires': meta['charm-metadata'].get('Requires'),
+        'provides': charm_metadata.get('Provides'),
+        'requires': charm_metadata.get('Requires'),
         'resources': _extract_resources(ref, meta.get('resources', {})),
         'revision_list': revision_list,
         'revision_number': ref.revision,
         'revisions': revisions,
         'series': meta.get('supported-series', {}).get('SupportedSeries'),
-        'tags': meta.get('Tags') or meta['charm-metadata'].get('Categories'),
+        'tags': charm_metadata.get('Tags') or charm_metadata.get('Categories'),
```

Comment:
Can you add a comment as to when it would be one or the other plz.

Reply:
Done. It's possible that we can remove the categories fallback at some point as I think categories are no longer a thing.